queries/lua/highlights.scm

```diff
@@ -2,25 +2,31 @@
 
 ;;; Builtins
 ;; Keywords
+(
+  [
+  "if"
+  "then"
+  "else"
+  "elseif"
+   ] @conditional
+  "end" @confitional
+)
 
-[
-"if"
-"then"
-"else"
-"elseif"
- ] @conditional
 
-[
-"do"
-"while"
-"repeat"
-"for"
-"in"
-] @repeat
+(
+  [
+  "do"
+  "while"
+  "repeat"
+  "for"
+  ] @repeat
+
```

Comment:
not (_)* between keyword and end?

Reply:
No need for this because of the semantic of this kind of matches : it matches sibling node, thus if if will match both anyway.